### PR TITLE
copy jwt claim to header

### PR DESCRIPTION
**Pull request by aryan16**
Updated the api to support the newly added feature in envoy - https://github.com/envoyproxy/envoy/pull/23444

### CODE REVIEW

File: security/v1beta1/jwt.proto [hunk 1]
```diff
@@ -155,3 +168,13 @@ message JWTHeader {
...
   string prefix = 2;
 }
+
+// This message specifies the detail for copying claim to header.
+message ClaimToHeader {
+  // The name of the header to be created. The header will be overridden if it already exists in the request.
+  string header = 1;
+
+  // The name of the claim to be copied from. Only claim of type string/int/bool is supported.
+  // The header will be set to an empty value if the claim does not exist or the type of the claim is not supported.
```
Comment (howardjohn) [inline security/v1beta1/jwt.proto:178]: Do we want to set empty or not set the header at all?

File: security/v1beta1/jwt.proto [hunk 2]
```diff
@@ -143,6 +143,19 @@ message JWTRule {
...
+
+  // This field specifies a list of operations to copy the claim to HTTP headers on a successfully verified token.
+  // This differs from the `output_payload_to_header` by allowing outputting individual claims instead of the whole payload.
+  // The header specified in each operation in the list must be unique.
+  // ```
+  //   outputClaimToHeaders:
+  //   - header: x-my-company-jwt-group
+  //     claim: my-group
+  //   - header: x-test-environment-flag
+  //     claim: test-flag
```
Comment (howardjohn) [inline security/v1beta1/jwt.proto:156]: are nested expressions supported?

File: security/v1beta1/jwt.proto (hunk 1 above)
Comment (aryan16) [reply security/v1beta1/jwt.proto:178]: oh yes, it won't set anything if claim is invalid. I will update the language.

File: security/v1beta1/jwt.proto (hunk 2 above)
Comment (aryan16) [reply security/v1beta1/jwt.proto:156]: yes, will add an example for that. Thanks

**Review by costinm — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**